test/test_util.jl

```diff
@@ -14,20 +14,37 @@ at input point `x` to confirm that there are correct ChainRules provided.
 - `f`: Function for which the `frule` and `rrule` should be tested.
 - `x`: input at which to evaluate `f` (should generally be set to an arbitary point in the domain).
 
-All keyword arguments except for `fdm` are passed to `isapprox`.
+- `test_wirtinger`: test whether the wirtinger derivative is correct, too
+
+All keyword arguments except for `fdm` and `test_wirtinger` are passed to `isapprox`.
 """
-function test_scalar(f, x; rtol=1e-9, atol=1e-9, fdm=_fdm, kwargs...)
+function test_scalar(f, x; rtol=1e-9, atol=1e-9, fdm=_fdm, test_wirtinger=x isa Complex, kwargs...)
     @testset "$f at $x, $(nameof(rule))" for rule in (rrule, frule)
         res = rule(f, x)
         @test res !== nothing  # Check the rule was defined
         fx, ∂x = res
         @test fx == f(x)  # Check we still get the normal value, right
 
         # Check that we get the derivative right:
-        @test isapprox(
-            ∂x(1), fdm(f, x);
-            rtol=rtol, atol=atol, kwargs...
-        )
+        if !test_wirtinger
+            @test isapprox(
+                ∂x(1), fdm(f, x);
+                rtol=rtol, atol=atol, kwargs...
+            )
+        else
+            # For complex arguments, also check if the wirtinger derivative is correct
+            ∂Re, ∂Im = fdm.((ϵ -> f(x + ϵ), ϵ -> f(x + im*ϵ)), 0)
```

Comment:
style suggestion:

```suggestion
            ∂Re = fdm(ϵ -> f(x + ϵ), 0)
            ∂Im = fdm(ϵ -> f(x + im*ϵ), 0)
```

Reply:
also, do we want `0` -> `zero(x)`?

Reply:
It should definitely always be real, so maybe `zero(real(x))` to support different float precision? Then again, do we even want to do fdm with anything less precise than `Float64`?

Reply:
cool - perhaps just `0.0` is fine then :)

Reply:
We should use `0` and let FiniteDifferences sort out promotion.